app/services/integration/keycloak_service.rb

```diff
@@ -38,7 +38,7 @@ def self.call(entry)
     attr_reader :tenant, :client_id, :scope
 
     def initialize(entry)
-      @client_id = entry.data&.dig('client_id')
+      @client_id = (entry.data || entry.previous_data)&.dig('client_id')
```

Comment:
We can fix the nil check 😄 
```suggestion
      @client_id = (entry.data || entry.previous_data || {}).dig('client_id')
```

Reply:
And maybe we should even add a method in `entry` for this 😄 
![image](https://user-images.githubusercontent.com/11318903/54608662-7a873b00-4a51-11e9-97f7-b72958b9c74f.png)

Maybe something like:
```ruby
class Entry
  def last_known_client_id
    (data || previous_data || {}).dig('client_id')
  end
end
```
And here: `entry.last_known_client_id`

Reply:
Entry is generic for any type as zync can sync more than just Applications. So Entry should not really know about `client_id` (but surely can know about previous data).

Reply:
Ok, then maybe:
```ruby
class Entry
  def last_known_data
    (data || previous_data || {})
  end
end
```
and then here:
`entry.last_known_data.dig('client_id')`